front-end/src/ui-components/VoteProgress.tsx

```diff
@@ -0,0 +1,108 @@
+// Copyright 2019-2020 @paritytech/polkassembly authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+import styled from '@xstyled/styled-components';
+import BN from 'bn.js';
+import { Progress } from 'semantic-ui-react';
+import React from 'react';
+
+import { chainProperties } from 'src/global/networkConstants';
+import getNetwork from 'src/util/getNetwork';
+import formatBnBalance from 'src/util/formatBnBalance';
+
+interface Props {
+	ayeVotes: BN,
+	className?: string,
+	nayVotes: BN,
+	passingThreshold: BN,
+	totalVotes: BN,
+}
+
+const bnToIntBalance = function (bn: BN): number{
+	return  Number(formatBnBalance(bn, { numberAfterComma: 2, withThousandDelimitor: false }));
+};
+
+const bnToStringBalanceDelimitor = function (bn: BN): string{
+	return  formatBnBalance(bn, { numberAfterComma: 2, withThousandDelimitor: true });
+};
+
+const VoteProgress = ({ ayeVotes, className, nayVotes, passingThreshold, totalVotes }: Props) => {
+	const network = getNetwork();
+	const tokenSymbol = chainProperties[network].tokenSymbol;
+
+	const nayVotesNumber = bnToIntBalance(nayVotes);
+	const passingThresholdNumber = bnToIntBalance(passingThreshold);
+	const isPassing = passingThreshold.lt(ayeVotes);
+	const ayeVotesNumber = bnToIntBalance(ayeVotes);
+	const totalVotesNumber = bnToIntBalance(totalVotes);
+	const passingThresholdPercent = isPassing
+		? (passingThresholdNumber/totalVotesNumber)*100
+		: ((passingThresholdNumber)/(passingThresholdNumber+nayVotesNumber))*100;
```

Comment:
why is `passingThresholdPercent` calculated differently depending on whether the threshold is met or not? `(passingThresholdNumber/totalVotesNumber)*100`sounds correct to me.

Reply:
Valid question. If it's passing, it means the threashold has been reached, the amount of aye/nays is ok.

If it's not passing, it means that you need more ayes, and if you need more, it means the `totalvotesNumber` will also change. But the `totalvotesNumber` that we have is what it is.

Example: there's 100 nays, 1 aye, the amount of nayes is say 150. With the first equation the threshold percentage is 150/101 x 100 ~= 150% If you use `left: 150%` in css, the threshold is floating outside of the progress bar. So I need to bring it back by calculating the real amount of total vote if we add the passingThreshold (ayes needed) and the nays.

Reply:
Ah, that makes sense. Thanks for explaining